src/statsd_client.cpp

```diff
@@ -84,6 +94,13 @@ int StatsdClient::init()
         return -1;
     }
 
+    // Don't block on stats sends....
+    int flags = fcntl(d->sock, F_GETFL, 0);
+    if (fcntl(d->sock, F_SETFL, flags | O_NONBLOCK) == -1) {
+      snprintf(d->errmsg, sizeof(d->errmsg), "could not enabled non blocking IO, err=%m");
+      return -1;
+    }
+
```

Comment:
That is not that useful on its own. The code that does the writing must also be modified to handle the situation where it will receive an E_AGAIN. Otherwise you will send part of a message then break out of writing because you would block. You will have to write more code to either continue later or ...